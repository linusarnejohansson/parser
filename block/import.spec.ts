import { Error, IO } from "@cogneco/mend"
import { Source } from "../Source"
import { parse } from "."

describe("block.import", () => {
	it("simple", () => {
		const result = parse(new Source(IO.StringReader.create("!import ./example/subdocument\n"), new Error.ConsoleHandler())) || []
		expect(result.map(node => node.toObject())).toMatchSnapshot()
	})
})
