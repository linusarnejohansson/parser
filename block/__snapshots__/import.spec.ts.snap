// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`block.import simple 1`] = `
Array [
  Object {
    "class": "Block.Import",
    "content": Object {
      "class": "File",
      "content": Array [
        Object {
          "class": "Block.Heading",
          "content": Array [
            Object {
              "class": "Inline.Text",
              "value": "Header",
            },
          ],
          "level": 1,
        },
        Object {
          "class": "Block.Paragraph",
          "content": Array [
            Object {
              "class": "Inline.Text",
              "value": "This is a paragraph.
",
            },
          ],
        },
      ],
    },
    "source": "./example/subdocument",
  },
]
`;
